libslide/src/bignum.rs

```diff
@@ -0,0 +1,100 @@
+use std::fmt;
+const DIGIT_COUNT: usize = 10000;
+
+// this probably can go in utils but I put it in here for now
+trait StringUtils {
+    fn substring(&self, start: usize, len: usize) -> Self;
+}
+
+impl StringUtils for String {
+    fn substring(&self, start: usize, len: usize) -> Self {
+        self.chars().skip(start).take(len).collect()
+    }
+}
```

Comment:
👍 

I wonder if you could do something like

```suggestion
impl StringUtils for String {
    fn substring(&self, start: usize, len: usize) -> Self {
        self[start..start+len].to_owned()
    }
}
```

?

Reply:
no no no. Rust does not allow subscript operator for strings.

Reply:
What about slices? https://doc.rust-lang.org/book/ch04-03-slices.html#string-slices

Reply:
They don’t work Bc slices are byte based and we need character based.

Reply:
They don’t work Bc slices are byte based and we need character based.

Reply:
ahh good call